src/transformers/diffLineTransformer.js

```diff
@@ -0,0 +1,47 @@
+// @ts-check
+const { highlightDiffLine } = require('./transformerUtils');
+
+function createDiffLineTransformer() {
+  // How do I get the parsed code fence options here?
+  /** @type {LineTransformer<any>} */
+  const diffLineTransformer = ({ line, meta: { diff } }) => {
+    if (diff && line.text.startsWith('+ ')) {
+      line.text = line.text.replace(/^\+ /, '');
+      return {
+        line: highlightDiffLine(line, 'add'),
+        data: { isDiffAdd: true },
+        gutterCells: [
+          {
+            className: 'grvsc-diff-add',
+            text: `+`
+          }
+        ]
+      };
+    }
+    if (diff && line.text.startsWith('- ')) {
+      line.text = line.text.replace(/^\- /, '');
+      return {
+        line: highlightDiffLine(line, 'del'),
+        data: { isDiffDel: true },
+        gutterCells: [
+          {
+            className: 'grvsc-diff-del',
+            text: `-`
+          }
+        ]
+      };
+    }
+    return { line };
+  };
+
+  diffLineTransformer.displayName = 'diff';
+  diffLineTransformer.schemaExtension = `
+    type GRVSCTokenizedLine {
+      isDiff: Boolean
```

Comment:
This should match what’s in `data`, so looks like it should be

```suggestion
      isDiffAdd: Boolean
      isDiffDel: Boolean
```

or maybe an enum would be better:

```graphql
enum GRVSCDiff {
  ADD
  DEL
}

type GRVSCTokenizedLine {
  diff: GRVSCDiff
}
```

(and update the `data` property with the literal string `"ADD"` or `"DEL"`)

Reply:
Hmm, maybe I can also try to generate TypeScript typings from these extensions so it can be validated properly...

Reply:
I added the enum.